hack/manifest-gen/assets/charts/eck/profile-restricted.yaml

```diff
@@ -1,7 +1,6 @@
-config:
-  createClusterScopedResources: false
+managedNamespaces: ["elastic-system"]
 
-  managedNamespaces: ["elastic-system"]
```

Comment:
I'm curious about the rational behind moving things that were in `config` at an upper-level. In my mind it feels natural to have all operator config items that can be customized though a flag or config map at the same level?

Reply:
It's a tough balance to strike. Managed namespaces are fundamental to how the operator works and the value is used in quite a few places to change how the templates are rendered. Therefore I thought it belonged at the top level.

The problem with moving all operator flags under `config` is that  sometimes just setting the config value is not enough. We need to do other things based on the value of that. Webhook is a good example of this. Just setting `enable-webhook` is not enough because you then have to make decisions about whether to use user-provided certificates, cert-manager, or operator-managed certificates and generate the appropriate manifests that require further customisation. That raises the question of where those additional customisation options should go. If they also go under `config`, we dilute the meaning of it. If we move the flag under `config` and the rest of the customisation somewhere else, the relationship between them won't be obvious to users.

Helm best practices document recommends not deeply nesting the values so the approach I used in this refactoring is to think in terms of major behavioural changes. Setting managed namespaces changes how the operator behaves in a significant way, so it merited being at the top level. The same logic applies to the webhook and so on. Hope that makes sense.

Reply:
Sounds good, thanks for the clarification!